### ISSUE TITLE
CRS not associated with geometries?

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/Shapefile.jl#114]
When loading a Shapefile with defined CRS, the individual geometries always have `GI.crs(geom) === nothing`, even though `GI.crs(table) isa GeoFormatTypes.ESRIWellKnownText`.  Would this need some form of refactor to store CRS in geometry as well?

**Comment by rafaqz** [comment]
Yes, there are no crs fields in the objects here. We could add them to most, but probably not to points.

And I guess strings are not actually copied/inlined for every polygon they are attached to? Its just a pointer to the same string underneath?

**Comment by asinghvi17** [comment]
I _think_ so? Strings are heap allocated, so presumably yes, and `Base.summarysize` doesn't tell you whether there's a pointer there or not.

**Comment by asinghvi17** [comment]
Yes, I just checked and the pointer to the strings is the same for multiple geometries initialized with the same CRS object.

**Comment by asinghvi17** [comment]
Maybe it's a bad idea, but we could just wrap every Shapefile geometry as a GI geometry when they're returned from `Handle`....

**Comment by rafaqz** [comment]
Hmm maybe that will just work? But its a weird nested object in terms of default types. I think adding a field is better.

**Comment by asinghvi17** [comment]
I guess we would add a `crs` argument to all `read` functions in Shapefile.jl?

**Comment by rafaqz** [comment]
Ah right that's kind of annoying but yes probably
